File: leetcodecpp/0001-Two Sum/2.cpp
#include <unordered_map>
#include <vector>
using namespace std;

class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        unordered_map<int, size_t> N;
        vector<int> res;
        for (size_t i = 0; i < nums.size(); ++i)
        {
            int num = nums[i];
            if (N.count(target - num))
            {
                res.push_back(i);
                res.push_back(N[target - num]);
            }
            else
            {
                N.insert({num, i});
            }
        }
        return res;
    }
};

File: leetcoderust/no_1342/src/2.rs
impl Solution {
    pub fn number_of_steps(num: i32) -> i32 {
        if num == 0 {
            return 0;
        } else {
            return if num % 2 == 1 {
                1 + Solution::number_of_steps(num - 1)
            } else {
                1 + Solution::number_of_steps(num / 2)
            }
        }
    }
}